lib/webauthn/public_key.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require "webauthn/attestation_statement/fido_u2f/public_key"
+require "cose/key"
+require "cose/key/ec2"
+require "cose/algorithm"
+
+module WebAuthn
+  class PublicKey
+    def self.deserialize(public_key)
+      cose_key = if WebAuthn::AttestationStatement::FidoU2f::PublicKey.uncompressed_point?(public_key)
+                   # Gem version v1.11.0 and lower, used to behave so that Credential#public_key
+                   # returned an EC P-256 uncompressed point.
+                   #
+                   # Because of https://github.com/cedarcode/webauthn-ruby/issues/137 this was changed
+                   # and Credential#public_key started returning the unchanged COSE_Key formatted
+                   # credentialPublicKey (as in https://www.w3.org/TR/webauthn/#credentialpublickey).
+                   #
+                   # Given that the credential public key is expected to be stored long-term by the gem
+                   # user and later be passed as the public_key argument in the
+                   # AuthenticatorAssertionResponse.verify call, we then need to support the two formats.
+                   COSE::Key::EC2.new(
+                     alg: COSE::Algorithm.by_name("ES256").id,
+                     crv: 1,
+                     x: public_key[1..32],
+                     y: public_key[33..-1]
+                   )
+                 else
+                   COSE::Key.deserialize(public_key)
+                 end
```

Comment:
I don't think rubocop has a cop for this, which I would enable if supported, but I think I prefer not having indentation depend of the length of a variable name. Smells a bit weird to me that I would need to update ~20 lines if I decide to change the name of `cose_key` variable to `key` e.g.

What do you think about?

```diff
diff --git a/lib/webauthn/public_key.rb b/lib/webauthn/public_key.rb
index 65521bd..5fc2e99 100644
--- a/lib/webauthn/public_key.rb
+++ b/lib/webauthn/public_key.rb
@@ -8,26 +8,27 @@ require "cose/algorithm"
 module WebAuthn
   class PublicKey
     def self.deserialize(public_key)
-      cose_key = if WebAuthn::AttestationStatement::FidoU2f::PublicKey.uncompressed_point?(public_key)
-                   # Gem version v1.11.0 and lower, used to behave so that Credential#public_key
-                   # returned an EC P-256 uncompressed point.
-                   #
-                   # Because of https://github.com/cedarcode/webauthn-ruby/issues/137 this was changed
-                   # and Credential#public_key started returning the unchanged COSE_Key formatted
-                   # credentialPublicKey (as in https://www.w3.org/TR/webauthn/#credentialpublickey).
-                   #
-                   # Given that the credential public key is expected to be stored long-term by the gem
-                   # user and later be passed as the public_key argument in the
-                   # AuthenticatorAssertionResponse.verify call, we then need to support the two formats.
-                   COSE::Key::EC2.new(
-                     alg: COSE::Algorithm.by_name("ES256").id,
-                     crv: 1,
-                     x: public_key[1..32],
-                     y: public_key[33..-1]
-                   )
-                 else
-                   COSE::Key.deserialize(public_key)
-                 end
+      cose_key =
+        if WebAuthn::AttestationStatement::FidoU2f::PublicKey.uncompressed_point?(public_key)
+          # Gem version v1.11.0 and lower, used to behave so that Credential#public_key
+          # returned an EC P-256 uncompressed point.
+          #
+          # Because of https://github.com/cedarcode/webauthn-ruby/issues/137 this was changed
+          # and Credential#public_key started returning the unchanged COSE_Key formatted
+          # credentialPublicKey (as in https://www.w3.org/TR/webauthn/#credentialpublickey).
+          #
+          # Given that the credential public key is expected to be stored long-term by the gem
+          # user and later be passed as the public_key argument in the
+          # AuthenticatorAssertionResponse.verify call, we then need to support the two formats.
+          COSE::Key::EC2.new(
+            alg: COSE::Algorithm.by_name("ES256").id,
+            crv: 1,
+            x: public_key[1..32],
+            y: public_key[33..-1]
+          )
+        else
+          COSE::Key.deserialize(public_key)
+        end
 
       new(cose_key: cose_key)
     end
```

Reply:
Oh, I see.

Didn't know that rubocop auto-corrected to that... :-/

Thanks for the update